### PR TITLE
Adapt plugin to work with alertManager v2 api

### PR DESCRIPTION
**Pull request by upcreid**
- Add an alert json example
- Change const to stick to a v2 api
- In api/v2 alert is a list of alerts, so we adapt the func statuses()
- Remove (comment) self.receiver check, not used anymore

### CODE REVIEW

File: defcon/plugins/test_alertmanager.py [hunk 1]
```diff
@@ -24,8 +24,8 @@ class AlertmanagerPluginTests(test.TestCase):
     """Test the plugins."""
 
     _API_URLS = [
-        'http://foo.alertmanager.io/api/v1/',
-        'http://foo.alertmanager0.14.io/api/v1/',
+        'http://foo.alertmanager.io/',
+        # 'http://foo.alertmanager0.14.io/',
```
Comment (uepoch) [inline defcon/plugins/test_alertmanager.py]: Remove instead of comment

File: defcon/plugins/alertmanager.py [hunk 2]
```diff
@@ -109,16 +135,18 @@ def render(template, data):
...
             return None
         status = alert.get('status', {})
-        if self.receiver is not None:
-            if 'routeOpts' in block:
-                if block['routeOpts']['receiver'] != self.receiver:
-                    return None
+
+        # if self.receiver is not None:
+        #     if 'routeOpts' in block:
+        #         if block['routeOpts']['receiver'] != self.receiver:
```
Comment (uepoch) [inline defcon/plugins/alertmanager.py]: Is there a reason to remove this code ?
Also, don't comment code

**Review by uepoch — CHANGES_REQUESTED** [review]

File: defcon/plugins/test_alertmanager.py (hunk 1 above)
Comment (upcreid) [reply defcon/plugins/test_alertmanager.py]: ok

File: defcon/plugins/alertmanager.py (hunk 2 above)
Comment (upcreid) [reply defcon/plugins/alertmanager.py]: old api was like:
```
...
            "receiver": "default",
            "groupBy": [
              "service",
              "alertname",
              "cluster"
            ],
            "groupWait": 30000000000,
            "groupInterval": 300000000000,
            "repeatInterval": 10800000000000
          },
          "alerts": [
...
```
i'll remove

File: defcon/plugins/alertmanager.py [hunk 3]
```diff
@@ -109,16 +135,23 @@ def render(template, data):
...
         logging.debug('Handling %s' % (alert))
         if not self.match_labels(alert['labels'], self.labels):
             return None
         status = alert.get('status', {})
+
+        # Check Receiver
         if self.receiver is not None:
-            if 'routeOpts' in block:
-                if block['routeOpts']['receiver'] != self.receiver:
+            if alert.get('receivers') is not None:
```
Comment (uepoch) [inline defcon/plugins/alertmanager.py:147]: If no receivers are being assigned to an alert, it shouldn't be evaluated by defcon ( alert['receivers'] == None)

File: defcon/plugins/alertmanager.py [hunk 4]
```diff
@@ -109,16 +135,23 @@ def render(template, data):
...
+                receiverList = []
+                for receveir in alert.get("receivers"):
```
Comment (uepoch) [inline defcon/plugins/alertmanager.py:149]: typo

File: defcon/plugins/test_alertmanager.py [hunk 5]
```diff
@@ -50,6 +49,10 @@ def test_all(self):
                 self.assertEqual(status['title'], 'ExampleAlertAlwaysFiring')
                 self.assertEqual(status['defcon'], 3)
 
+                # statuses[1] have wrong receiver, so it should raise an exeption
+                with self.assertRaises(IndexError):
```
Comment (uepoch) [inline defcon/plugins/test_alertmanager.py:53]: Test is a bit weird, but ok :P

**Review by uepoch — APPROVED** [review]
I will not block the PR anymore as I think the mandatory problems have been dealed with

If you want OPT: fix the None on receivers

If you want we can do it, then release to pipy the next release

**Pull request merged**